File: 122-best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxi = 0;
        int cp = prices[0];
        for(int i=1; i<prices.size(); i++){
            if(prices[i] > cp){
                maxi = maxi + (prices[i] - cp);
            }
            cp = prices[i];
        }
        return maxi;
    }
};

File: 122-best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxi = 0
        lsb = prices[0]
        for idx in range(1, len(prices)):
            if prices[idx] > prices[idx-1]:
                maxi = maxi + prices[idx] - prices[idx-1]
            
        return maxi
        